src/main/frontend/app/containers/ProductForm.js

```diff
@@ -0,0 +1,196 @@
+import React, { Component } from 'react';
+import FieldInput from '../components/FieldInput';
+
+class ProductForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errors: {},
+      product: {
+        name: '',
+        price: '',
+        description: '',
+        url: '',
+        imageUrl: ''
+      }
+    }
+
+    this.onChange = this.onChange.bind(this);
+    this.persistProduct = this.persistProduct.bind(this);
+    this.clearForm = this.clearForm.bind(this);
+    this.validateNameSelection = this.validateNameSelection.bind(this);
+    this.validatePriceSelection = this.validatePriceSelection.bind(this);
+    this.validateDescriptionSelection = this.validateDescriptionSelection.bind(this);
+    this.validateUrlSelection = this.validateUrlSelection.bind(this);
+    this.validateImageUrl = this.validateImageUrl.bind(this);
+  }
+
+  onChange(event) {
+    fieldName = event.target.name;
+    fieldValue = event.target.value;
+    this.setState({
+      [product.fieldName]: fieldValue
+    })
+  }
+
+  persistProduct(event) {
+    event.preventDefault();
+    console.log('hi')
+
+    this.clearForm(event);
+  }
+
+  validateNameSelection(selection) {
+    const { errors } = this.state;
+
+    if (selection.trim() === '') {
+      let newError = { nameSelected: 'You must enter a name for the product' }
+      this.setState({ errors: Object.assign({}, errors, newError) })
+      return false;
+    } else {
+      let errorState = errors;
+      delete errorState.nameSelected;
+      this.setState({ errors: errorState })
+      return true;
+    }
+  }
+
+  validatePriceSelection(selection) {
+    const { errors } = this.state;
+
+    if (selection.trim() === '') {
+      let newError = { priceSelected: 'You must enter a price for the product' }
+      this.setState({ errors: Object.assign({}, errors, newError) })
+      return false;
+    } else {
+      let errorState = errors;
+      delete errorState.priceSelected;
+      this.setState({ errors: errorState })
+      return true;
+    }
+  }
+
```

Comment:
The validate methods could likely be refactored into one method with a conditional inside. 
Consider this for future branches.